delete_version_after_test.go

```diff
@@ -0,0 +1,265 @@
+				if got != nil {
+					t.Fatalf("mount value: delete_version_after %#v, want no delete_version_after", got)
+				}
+			} else {
+				want, got := tt.mount.String(), resp.Data["delete_version_after"]
+				if want != got {
+					t.Fatalf("want delete_version_after: %v, got %v", want, got)
+				}
+			}
+
+			data = map[string]interface{}{
+				"max_versions":         2,
+				"delete_version_after": tt.meta.String(),
+			}
+			req = &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "metadata/foo",
+				Storage:   storage,
+				Data:      data,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantNoResponse(t, resp, err)
+
+			req = &logical.Request{
+				Operation: logical.ReadOperation,
+				Path:      "metadata/foo",
+				Storage:   storage,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantResponse(t, resp, err)
+			want, got := tt.meta.String(), resp.Data["delete_version_after"]
+			if want != got {
+				t.Fatalf("want delete_version_after: %v, got %v", want, got)
+			}
+
+			data = map[string]interface{}{
+				"data": map[string]interface{}{
+					"bar": "baz1",
+				},
+			}
+			req = &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "data/foo",
+				Storage:   storage,
+				Data:      data,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantResponse(t, resp, err)
+			if !tt.wantDeletionTime {
+				if dtv := resp.Data["deletion_time"].(string); dtv != "" {
+					t.Logf("resp: %#v", resp)
+					t.Fatalf("deletion_time %#v, want no deletion_time", dtv)
+				}
+			} else {
+				got := lifetime(t, resp.Data)
+				if tt.want != got {
+					t.Fatalf("diff between deletion_time and created_time %v, want %v", got, want)
+				}
+			}
+
+			req = &logical.Request{
+				Operation: logical.ReadOperation,
+				Path:      "data/foo",
+				Storage:   storage,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantResponse(t, resp, err)
+			meta := resp.Data["metadata"].(map[string]interface{})
+			if !tt.wantDeletionTime {
+				if dtv := meta["deletion_time"].(string); dtv != "" {
+					t.Logf("meta: %#v", meta)
+					t.Fatalf("deletion_time %#v, want no deletion_time", dtv)
+				}
+			} else {
+				got := lifetime(t, meta)
+				if tt.want != got {
+					t.Fatalf("diff between deletion_time and created_time %v, want %v", got, want)
+				}
+			}
+
+			data = map[string]interface{}{
+				"versions": "1",
+			}
+			req = &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "delete/foo",
+				Storage:   storage,
+				Data:      data,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantNoResponse(t, resp, err)
+
+			data = map[string]interface{}{
+				"versions": "1",
+			}
+			req = &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "undelete/foo",
+				Storage:   storage,
+				Data:      data,
+			}
+			undeleteTime := time.Now()
```

Comment:
Might want to add a comment explaining that after an undelete the deletion timer is reset.

Reply:
Good idea.